src/Algebra/Module/Properties.agda

```diff
@@ -0,0 +1,97 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of algebraic structures defined over some other
+-- structure, like modules and vector spaces
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Module.Properties where
+
+open import Algebra
+open import Algebra.Module
+open import Level
+
+private
+  variable
+    c ℓ : Level
+
+self-leftSemimodule : (R : Semiring c ℓ) → LeftSemimodule R c ℓ
+self-leftSemimodule semiring = record
+  { Carrierᴹ = Carrier
+  ; _≈ᴹ_ = _≈_
+  ; _+ᴹ_ = _+_
+  ; _*ₗ_ = _*_
+  ; 0ᴹ = 0#
+  ; isLeftSemimodule = record
+    { +ᴹ-isCommutativeMonoid = +-isCommutativeMonoid
+    ; *ₗ-cong = *-cong
+    ; *ₗ-zeroˡ = zeroˡ
+    ; *ₗ-distribʳ = distribʳ
+    ; *ₗ-identityˡ = *-identityˡ
+    ; *ₗ-assoc = *-assoc
+    ; *ₗ-zeroʳ = zeroʳ
+    ; *ₗ-distribˡ = distribˡ
+    }
+  } where open Semiring semiring
+
+self-rightSemimodule : (R : Semiring c ℓ) → RightSemimodule R c ℓ
+self-rightSemimodule semiring = record
+  { Carrierᴹ = Carrier
+  ; _≈ᴹ_ = _≈_
+  ; _+ᴹ_ = _+_
+  ; _*ᵣ_ = _*_
+  ; 0ᴹ = 0#
+  ; isRightSemimodule = record
+    { +ᴹ-isCommutativeMonoid = +-isCommutativeMonoid
+    ; *ᵣ-cong = *-cong
+    ; *ᵣ-zeroʳ = zeroʳ
+    ; *ᵣ-distribˡ = distribˡ
+    ; *ᵣ-identityʳ = *-identityʳ
+    ; *ᵣ-assoc = *-assoc
+    ; *ᵣ-zeroˡ = zeroˡ
+    ; *ᵣ-distribʳ = distribʳ
+    }
+  } where open Semiring semiring
+
+self-semimodule : (R : CommutativeSemiring c ℓ) → Semimodule R c ℓ
+self-semimodule commutativeSemiring = record
+  { isSemimodule = record
+    { isLeftSemimodule =
+      let open LeftSemimodule (self-leftSemimodule semiring) in
+      isLeftSemimodule
+    }
+  } where open CommutativeSemiring commutativeSemiring
+
+self-leftModule : (R : Ring c ℓ) → LeftModule R c ℓ
+self-leftModule ring = record
+  { -ᴹ_ = -_
+  ; isLeftModule = record
+    { isLeftSemimodule =
+      let open LeftSemimodule (self-leftSemimodule semiring) in
```

Comment:
This inline let binding is rather horrifying... can't you move this to the where clause?